features/git-rename-branch/feature_branch/with_remote_origin/with_tracking_branch/with_open_changes.feature

```diff
@@ -0,0 +1,34 @@
+Feature: git rename-branch: renaming a feature branch with a tracking branch (with open changes)
+
+  As a developer with a poorly named feature branch
+  I want to be able to rename it safely in one easy step
+  So that I can stay organized and remain productive
+
+
+  Background:
+    Given I have a feature branch named "current-feature"
+    And the following commits exist in my repository
+      | BRANCH          | LOCATION         | MESSAGE        |
+      | main            | local and remote | main commit    |
+      | current-feature | local and remote | feature commit |
+    And I am on the "current-feature" branch
+    And I have an uncommitted file
+    When I run `git rename-branch current-feature renamed-feature`
+
+
+  Scenario: result
+    Then it runs the Git commands
+      | BRANCH          | COMMAND                                         |
+      | current-feature | git fetch --prune                               |
+      |                 | git stash -u                                    |
+      |                 | git checkout -b renamed-feature current-feature |
```

Comment:
Yeah, the reason I didn't use `-m` is because I think it would make it more complicated and asymmetric since the remote branches don't have a `-m` equivalent. It also lets this reuse existing code, and doesn't have a significant difference in performance.